travis/self_tests

```diff
@@ -1,77 +1,406 @@
+
+    def test_get_modules_changed(self):
+        """Testing git run from getaddons"""
+        self.assertIsNotNone(
+            getaddons.get_modules_changed(self.repo_dir))
+
+    @unittest.skipIf(os.environ.get('LINT_CHECK', 0) != '1', "Set LINT_CHECK")
+    def test_check_vmaster(self):
+        """PYLINT_EXPECTED_ERRORS="89"
+           VERSION="master"
+        """
+        print "VERSION=MASTER"
+        expected_errors = EXPECTED_ERRORS.copy()
+        expected_errors.update(EXPECTED_ERRORS_PR)
+        self.pylint_run(
+            is_pr=True, version="master",
+            expected_errors=91, errors_dict=expected_errors)
+
+        expected_errors = EXPECTED_ERRORS.copy()
+        self.pylint_run(
+            is_pr=False, version="master",
+            expected_errors=27, errors_dict=expected_errors)
+
+    @unittest.skipIf(os.environ.get('LINT_CHECK', 0) != '1', "Set LINT_CHECK")
+    def test_check_v7(self):
+        """PYLINT_EXPECTED_ERRORS="84"
+           VERSION="7.0"
+        """
+        print "VERSION=7.0"
+        expected_errors = EXPECTED_ERRORS.copy()
+        expected_errors.update(EXPECTED_ERRORS_PR)
+        expected_errors.pop('openerp-exception-warning')
+        expected_errors.pop('missing-readme')
+        expected_errors.pop('class-camelcase')
+        expected_errors.pop('manifest-deprecated-key')
+        expected_errors.pop('copy-wo-api-one')
+        expected_errors['manifest-version-format'] = 5
+        self.pylint_run(
+            is_pr=True, version="7.0",
+            expected_errors=84, errors_dict=expected_errors)
+
+        expected_errors = EXPECTED_ERRORS.copy()
+        self.pylint_run(
+            is_pr=False, version="7.0",
+            expected_errors=27, errors_dict=expected_errors)
+
+    @unittest.skipIf(os.environ.get('LINT_CHECK', 0) != '1', "Set LINT_CHECK")
+    def test_check_v8(self):
+        print "VERSION=8.0"
+        expected_errors = EXPECTED_ERRORS.copy()
+        expected_errors.update(EXPECTED_ERRORS_PR)
+        self.pylint_run(
+            is_pr=True, version="8.0",
+            expected_errors=91, errors_dict=expected_errors)
+        expected_errors = EXPECTED_ERRORS.copy()
+        self.pylint_run(
+            is_pr=False, version="8.0",
+            expected_errors=27, errors_dict=expected_errors)
+
+    @unittest.skipIf(os.environ.get('LINT_CHECK', 0) != '1', "Set LINT_CHECK")
+    def test_check_without_version(self):
+        """PYLINT_EXPECTED_ERRORS="27"
+           VERSION=""
+        """
+        print "VERSION=''"
+        expected_errors = EXPECTED_ERRORS.copy()
+        expected_errors.update(EXPECTED_ERRORS_PR)
+        self.pylint_run(
+            is_pr=True, version="",
+            expected_errors=91, errors_dict=expected_errors)
+        expected_errors = EXPECTED_ERRORS.copy()
+        self.pylint_run(
+            is_pr=False, version="",
+            expected_errors=27, errors_dict=expected_errors)
+
+    def version_validate(self, version):
```

Comment:
what do you think  @moylop260 about https://github.com/Vauxoo/maintainer-quality-tools/pull/172#discussion_r101331699